### ISSUE TITLE
Review docs for clarity and accuracy (difficulty: easy)

### ISSUE THREAD

**Issue by CactusCompute** [cactus-compute/cactus#297]
Check if the docs are accurate, up to date and intuitive:

[docs](https://github.com/cactus-compute/cactus/tree/main/docs)

**Comment by Crater78** [comment]
I think they look very nice (what did you use to make them), but I think definitions for each method could be beneficial.
(Kind of like what python does in [these](https://docs.python.org/3/library/math.html#math.radians) docs, where each method is defined.)

\- Crater78

**Comment by UsmanCh1226** [comment]
### Proposal: Documentation Improvement for clarity and accuracy

I would like to help with this issue. Based on the feedback from @Crater78, I propose updating the **Error Handling** and **Performance** sections with more specific definitions:

#### 1. Standardized Error Handling
Most functions return:
* **Positive values or 0**: Success (represents bytes written to the buffer).
* **Negative values**: Failure. Use `cactus_get_last_error()` for details.
  * `-1`: Invalid model/corpus path.
  * `-2`: Buffer size too small.
  * `-3`: Malformed JSON input.
  * `-4`: Unsupported operation for model type.

#### 2. Performance & Best Practices
* **Thread Safety**: `cactus_model_t` is **not thread-safe**. Possible one instance per thread.
* **Reuse Models**: Initialize once via `cactus_init()` and reuse the handle to avoid reload latency and improve efficiency.
* **KV Cache Tuning**: Use `cactus_reset()` to clear state between unrelated conversations.
* **Streaming**: Always use `cactus_token_callback` for responsive UI during long generations.

Please let me know if this looks good! 

- UsmanCh1226

**Comment by VamanR06** [comment]
It would be helpful if there was a table of contents in the docs to allow a developer to easily find a function they need without scrolling all the way through. Also, the examples in the root directory README could be moved to the specific module its from (engine example in cactus_engine.md). Would make it easier to find relevant code instead of searching elsewhere.
- VamanR06

**Comment by Shawaqfeh715** [comment]
Proposal for the Graph file documentation
 

I would like to make a suggestion for the `cactus_graph` file, I think it would be helpful in the documentation of the graph file to provide a brief latex section to explain how specific calculations especially the advanced ones like `layernorm` and `rope_output` work, because for readers unfamiliar with the topics they may find it out of context. 

The current documentation assumes deep familiarity with neural network operations and linear algebra. Functions like `layernorm`,`rope`, `attention`, `matmul`, and broadcasting cannot be used effectively by developers unfamiliar with the underlying mathematics. Which creates a struggle for developers who haven't specialized in ML trying to access the API and use it.

solution:

-a short description in english
-LaTeX formula
-shape transformation and how input maps to output shapes.
-explanation of the parameters.


- Shawaqfeh715

**Comment by HenryNdubuaku** [comment]
Go for it, don't make it chunky though.

**Comment by N00bcak** [comment]
Hmm. Have a few suggestions for [Graph API](https://github.com/cactus-compute/cactus/blob/main/docs/cactus_graph.md):

1. Relating Cactus ops (roughly) to NumPy/Torch ones
Add a one-liner per op approximation, so newer developers can get a rough idea of **what to google** first:
Examples:
- `softmax(x, axis) ~= torch.softmax(x, dim=axis)`
- `matmul(a, b, pretransposed_rhs) ~= torch.matmul(a, b) + <insert specific behavior due to pretransposed_rhs>`
- `gather(embeddings, indices) ~= torch.nn.functional.embedding(indices, embeddings)`
- `sample(logits, temperature, top_p, top_k) ~= standard top-k/top-p sampling` (include a link to a beginner-friendly explanation like [this](https://huggingface.co/blog/how-to-generate))

2. Quantization/precision contracts
A few questions remain unclear with the API. To minimize guessing we should answer these questions:
- do we do implicit casting when mixing INT4/INT8/FP16? If so, how & where? If not, why?
- how are matmul accumulates are performed (what precision, any scaling, any clamping etc.)

Also, remove or replace `set_quantization_scale` (appears not to be implemented in the code, but I could be wrong)

3. Debugging workflow (`execute("profile_output.json")`) exists, but docs should give more details, such as:
- what’s in the profiling JSON + how to identify bottlenecks
- how to dump intermediates (`get_output(node_id)`) and sanity-check shape/precision/broadcasting

LMK if y'all have any thoughts on this
- N00bcak
